scripts/arch-chroot-lite

```diff
@@ -22,8 +22,28 @@ chroot_setup() {
     # arch-chroot-lite drops the conditional bind mount on the chroot path, as
     # it seemed to shadow mounts set up before arch-chroot was invoked
 
+    # This directory stores secret GPG keys, so its contents must be kept secret
+    # at all costs.  Anyone with access to the files in it can compromise the
+    # built TemplateVM and all VMs based on it.
+    secret_key_dir="$1/etc/pacman.d/gnupg/private-keys-v1.d" &&
+
     # private-keys-v1.d does not exist before we create the tmpfs
-    mkdir -p "$1/etc/pacman.d/gnupg/private-keys-v1.d"
+    mkdir -p -m 0755 -- "${secret_key_dir%/*}" &&
+    mkdir -m 0000 -- "$secret_key_dir" &&
+
+    # Set the correct permissions for mount points
+    chmod -- 0755 "$1/dev" "$1/run" &&
+    chmod -- 0555 "$1/proc" "$1/sys" &&
+    chmod -- 1777 "$q/tmp" &&
+
+    # Mark private-keys-v1.d immutable, so that files (such as secret keys)
+    # cannot accidentally be created in it.  Also mark other directories
+    # immutable that should only be used as mount points.
+    chattr +i -- "$secret_key_dir" "$1/sys" "$1/dev" "$1/proc" "$1/run" "$1/tmp" &&
+
+    # Turn off swap system-wide, so that we do not accidentally leak secret keys
+    # to disk
+    swapoff -a &&
```

Comment:
No, this will likely cause the VM to crash on OOM - qmemman kind of depends on swap presence to have a little more time to react (take memory from other VMs). But also, you don't enable it afterwards, which is even worse.

Maybe use `ramfs` instead of `tmpfs` to avoid using swap?

Reply:
Good idea.  In the future, we should encrypt the swap volume, which will solve all of these problems.

Reply:
Fixed.